File: Gateway/Converter/Model/StatusConverter.cpp
#include "pch.h"
#include "StatusConverter.h"

Status StatusConverter::toNative(StatusDto^ dto)
{
	Status native;

	if (dto == nullptr) {
		return native;
	}

	native.Attack = dto->Attack;
	native.Hp = dto->HP;
	native.Defense = dto->Defense;
	native.Speed = dto->Speed;
	native.CriticalRate = dto->CritRate;
	native.CriticalDamage = dto->CritDamage;
	native.EffectHit = dto->EffectHit;
	native.EffectResist = dto->EffectResist;
	native.AdditionalAttackRate = dto->AdditionalAttackRate;
	native.AdditionalHpRate = dto->AdditionalHpRate;
	native.AdditionalDefenseRate = dto->AdditionalDefenseRate;

	return native;
}

StatusDto^ StatusConverter::toDto(const Status& native)
{
	StatusDto^ dto = gcnew StatusDto();

	dto->Attack = native.Attack;
	dto->HP = native.Hp;;
	dto->Defense = native.Defense;
	dto->Speed = native.Speed;
	dto->CritRate = native.CriticalRate;
	dto->CritDamage = native.CriticalDamage;
	dto->EffectHit = native.EffectHit;
	dto->EffectResist = native.EffectResist;
	dto->AdditionalAttackRate = native.AdditionalAttackRate;
	dto->AdditionalHpRate = native.AdditionalHpRate;
	dto->AdditionalDefenseRate = native.AdditionalDefenseRate;

	return dto;
}

File: Gui/Converter/StatusConverter.cs
using SaveData.Model;
using SaveData.Model.Development;

namespace Gui.Converter
{
	public static class StatusConverter
	{
		public static StatusSaveData ToSaveData(StatusDto dto)
		{
			if (dto == null)
			{
				return null;
			}

			return new StatusSaveData
			{
				Attack = dto.Attack,
				HP = dto.HP,
				Defense = dto.Defense,
				Speed = dto.Speed,
				CritRate = dto.CritRate,
				CritDamage = dto.CritDamage,
				EffectHit = dto.EffectHit,
				EffectResist = dto.EffectResist
			};
		}

		public static StatusDto ToDto(StatusSaveData saveData)
		{
			if (saveData == null)
			{
				return null;
			}

			return new StatusDto
			{
				Attack = saveData.Attack,
				HP = saveData.HP,
				Defense = saveData.Defense,
				Speed = saveData.Speed,
				CritRate = saveData.CritRate,
				CritDamage = saveData.CritDamage,
				EffectHit = saveData.EffectHit,
				EffectResist = saveData.EffectResist
			};
		}

		public static FullStatusSaveData ToFullSaveData(StatusDto dto)
		{
			return new FullStatusSaveData
			{
				Attack = dto.Attack,
				HP = dto.HP,
				Defense = dto.Defense,
				Speed = dto.Speed,
				CritRate = dto.CritRate,
				CritDamage = dto.CritDamage,
				EffectHit = dto.EffectHit,
				EffectResist = dto.EffectResist,
				AdditionalAttackRate = dto.AdditionalAttackRate,
				AdditionalHpRate = dto.AdditionalHpRate,
				AdditionalDefenseRate = dto.AdditionalDefenseRate
			};
		}
	}
}
